pandas/tests/test_algos.py

```diff
@@ -536,11 +536,9 @@ def test_objects(self):
         assert isinstance(result, np.ndarray)
 
     def test_object_refcount_bug(self):
-        lst = ["A", "B", "C", "D", "E"]
-        msg = "unique with argument that is not not a Series"
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            for i in range(1000):
-                len(algos.unique(lst))
+        lst = np.array(["A", "B", "C", "D", "E"], dtype=object)
```

Comment:
would this still catch the original refcount bug?

im guessing we'd get most of the performance improvement by passing check_stacklevel=False to assert_produces_warning

Reply:
Yeah looking back it appears this should have still happened if a numpy array is passed, so I think preemptively changing to address the FutureWarning too is OK